src/main/java/br/ufpe/cin/mergers/util/RenamingUtils.java

```diff
@@ -72,12 +72,16 @@ public static boolean hasDefinedBody(String methodSource) {
         return methodSource.contains("{");
     }
 
-    public static String getMethodBodyWithoutWhitespaces(FSTNode node) {
-        if (isMethodOrConstructorNode(node))
-            return StringUtils.deleteWhitespace(((FSTTerminal) node).getDeclarationBody());
-
-        return StringUtils.EMPTY;
-    }
+	public static String getMethodBodyWithoutWhitespaces(FSTNode node) {
+		if (isMethodOrConstructorNode(node)) {
+			String methodBody = StringUtils.deleteWhitespace(((FSTTerminal) node).getDeclarationBody());
```

Comment:
@guilhermejccavalcanti 

não é melhor verificar se o método é abstrato e, se for, retornar vazio? senão faz o que fazia antes.

digo isso porque um método concreto em Java pode ter ";" como corpo. olha Body em https://docs.oracle.com/javase/specs/jls/se7/html/jls-18.html

Reply:
> não é melhor verificar se o método é abstrato e, se for, retornar vazio? senão faz o que fazia antes.

Pra contextualizar, eu me refiro como método abstrato aqueles que não tem corpo (só o cabeçalho). Dái, teria alguma sugestão solução simples em mente para identificar um método abstrato? A keword "abstract" é óbvia, mas métodos de interface não possuem "abstract" e o corpo é vazio também. Neste caso, teria que usar algum outro recurso pra identificar de a classe é uma interface, e não me vem à cabeça soluções simples? Usar JDT ou varrer a árvore do FST parece-me "overkill".

Reply:
certo @guilhermejccavalcanti. podemos ter os seguintes casos

1. void m();
2. void m(){;}

no primeiro caso, getDeclarationBody deveria retornar "". no segundo caso, getDeclarationBody deveria retornar ";".

o s3m tá representando os dois casos com um nó que tem o mesmo conteúdo?

Reply:
> o s3m tá representando os dois casos com um nó que tem o mesmo conteúdo?
Não. O body de 1. é representado como ";" e o de 2. é representado como "{;}". Acredito então que podemos verificar a existência de "{" e "}" pra detectar o que é método abstrato.